src/python/pants/build_graph/target.py

```diff
@@ -473,7 +473,10 @@ def transitive_invalidation_hash(self, fingerprint_strategy=None, depth=0):
       def dep_hash_iter():
         for dep in self.dependencies:
           try:
-            dep_hash = dep.transitive_invalidation_hash(fingerprint_strategy, depth=depth+1)
+            if getattr(self, 'strict_deps', None):
```

Comment:
fingerprint strategy is on per target level without considering dependencies, thus I tend to try Stu's advice here if it's OK with you.